// ng2-formly/lib/templates.js
"use strict";
var formlyfield_checkbox_1 = require("./templates/formlyfield.checkbox");
exports.FormlyFieldCheckbox = formlyfield_checkbox_1.FormlyFieldCheckbox;
var formlyfield_multicheckbox_1 = require("./templates/formlyfield.multicheckbox");
exports.FormlyFieldMultiCheckbox = formlyfield_multicheckbox_1.FormlyFieldMultiCheckbox;
var formlyfield_input_1 = require("./templates/formlyfield.input");
exports.FormlyFieldInput = formlyfield_input_1.FormlyFieldInput;
var formlyfield_radio_1 = require("./templates/formlyfield.radio");
exports.FormlyFieldRadio = formlyfield_radio_1.FormlyFieldRadio;
var formlyfield_textarea_1 = require("./templates/formlyfield.textarea");
exports.FormlyFieldTextArea = formlyfield_textarea_1.FormlyFieldTextArea;
var formlyfield_select_1 = require("./templates/formlyfield.select");
exports.FormlyFieldSelect = formlyfield_select_1.FormlyFieldSelect;
var field_1 = require("./templates/field");
exports.Field = field_1.Field;
var templates_1 = require("./templates/templates");
exports.TemplateDirectives = templates_1.TemplateDirectives;
var formlyBootstrap_1 = require("./templates/formlyBootstrap");
exports.FormlyBootstrap = formlyBootstrap_1.FormlyBootstrap;
